bot/exts/utils/reminders.py

```diff
@@ -35,6 +37,149 @@
+        for duration in self.SNOOZE_DURATIONS:
+            self.dropdown.add_option(label=duration)
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        """Ensure that the user clicking the button is the member who invoked the command."""
+        if interaction.user.id != self.reminder_dct['author']:
+            await interaction.response.send_message(":x: This is not your reminder to snooze!", ephemeral=True)
+            return False
+        return True
+
+    @discord.ui.select(placeholder="Select the snooze duration")
+    async def select_snooze_duration(self, interaction: discord.Interaction, _: discord.ui.Select) -> None:
+        """Drop down menu which contains a list of snooze durations one can choose."""
+        selected_duration = interaction.data["values"][0]
+        current_datetime = datetime.now(timezone.utc)
+        if (num_seconds := self.SNOOZE_DURATIONS.get(selected_duration)) is None:
+            amount, unit = selected_duration.split()
+            if unit.startswith("month"):
+                num_seconds = (current_datetime + relativedelta(months=int(amount)) - current_datetime).total_seconds()
+
+            else:  # unit is years
+                num_seconds = (current_datetime + relativedelta(years=int(amount)) - current_datetime).total_seconds()
+
+        new_reminder_end_time = current_datetime + relativedelta(seconds=num_seconds)
+        expiry_dct = {'expiration': new_reminder_end_time.isoformat()}
+
+        # Reminder was snoozed, so we need to reschedule it
+        reminder_id = self.reminder_dct['id']
+        log.debug(f"Snoozing reminder #{reminder_id}. New expiration: {new_reminder_end_time}")
+
+        await self.reminders_instance.bot.api_client.patch(
+            f'bot/reminders/{reminder_id}',
+            json=expiry_dct
+        )
+
+        self.reminders_instance.scheduler.schedule_at(
+            new_reminder_end_time,
+            reminder_id,
+            self.reminders_instance.send_reminder(self.reminder_dct | expiry_dct)
+        )
+
+        new_end_timestamp = time.discord_timestamp(new_reminder_end_time, format=time.TimestampFormats.DAY_TIME)
+
+        new_embed = discord.Embed(
+            colour=discord.Colour.green(),
+            title=random.choice(POSITIVE_REPLIES),
+            description=f"Successfully snoozed reminder for {selected_duration}. New remind time: {new_end_timestamp}",
+        )
+        new_embed.set_footer(text=f"ID: {reminder_id}")
+        try:
+            await interaction.response.edit_message(embed=new_embed, view=None)
+        except discord.NotFound:
+            await interaction.message.channel.send(embed=new_embed)
+
+
+class SnoozeButtonView(discord.ui.View):
+    """The reminder's snooze button UI View."""
+
+    def __init__(self, reminders_instance: "Reminders", reminder_dct: dict):
+        super().__init__(timeout=30)  # user has 5 minutes from reminder to snooze it
```

Comment:
Adding this comment as a reminder to myself to fix this comment/timeout as is appropriate before this gets merged.